### PR TITLE
Add data caching functionality.

### PR DESCRIPTION
**Pull request by simonasvaitkus**

### CODE REVIEW

File: b_lambda_layer_common/source/python/b_lambda_layer_common/cache/data_cache.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        :param pointer: A string or a set of strings used to reference cached data.
+        :param func: A function whose return value will be cached.
+        :param args: Arguments that will be passed to func.
+
+        :key kwargs: Keyword arguments that will be passed to func.
+
+        :return: Cached data.
+        """
+        if self.should_refresh():
+            logger.info('Cached data should be reset.')
```
Comment (robertasstankevicius) [inline b_lambda_layer_common/source/python/b_lambda_layer_common/cache/data_cache.py:51]: When logging what is going to happen, use continuous present tense. Then if an error follows, you will know what action was being performed.
For example: Resetting cached data...

File: b_lambda_layer_common/source/python/b_lambda_layer_common/cache/data_cache.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    def use_cache(self, pointer: Union[str, Set[str]], func: Callable, *args, **kwargs) -> Any:
+        """
+        Cache function return values.
+
+        :param pointer: A string or a set of strings used to reference cached data.
+        :param func: A function whose return value will be cached.
+        :param args: Arguments that will be passed to func.
+
+        :key kwargs: Keyword arguments that will be passed to func.
```
Comment (robertasstankevicius) [inline b_lambda_layer_common/source/python/b_lambda_layer_common/cache/data_cache.py:46]: kwargs is also a parameter

File: b_lambda_layer_common/source/python/b_lambda_layer_common/cache/data_cache.py (hunk 1 above)
Comment (robertasstankevicius) [reply b_lambda_layer_common/source/python/b_lambda_layer_common/cache/data_cache.py:51]: Don't forget the three dots ...

**Review by robertasstankevicius — CHANGES_REQUESTED** [review]

**Review by robertasstankevicius — APPROVED** [review]

File: b_lambda_layer_common/source/python/b_lambda_layer_common/cache/data_cache.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+try:
+    from b_lambda_layer_common.exceptions.container.internal_error import InternalError
+    from b_lambda_layer_common.ssm.refreshable import Refreshable
+except ImportError:
+    logger.exception(f'Failed import.')
+    from b_lambda_layer_common.source.python.b_lambda_layer_common.exceptions.container.internal_error import InternalError
+    from b_lambda_layer_common.source.python.b_lambda_layer_common.ssm.refreshable import Refreshable
+
+
+class DataCache(Refreshable):
```
Comment (laimonassutkus) [inline b_lambda_layer_common/source/python/b_lambda_layer_common/cache/data_cache.py:15]: Rename DataCache to InMemoryDataCache.

File: b_lambda_layer_common/source/python/b_lambda_layer_common/cache/data_cache.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+logger = logging.getLogger(__name__)
+
+try:
+    from b_lambda_layer_common.exceptions.container.internal_error import InternalError
+    from b_lambda_layer_common.ssm.refreshable import Refreshable
+except ImportError:
+    logger.exception(f'Failed import.')
+    from b_lambda_layer_common.source.python.b_lambda_layer_common.exceptions.container.internal_error import InternalError
+    from b_lambda_layer_common.source.python.b_lambda_layer_common.ssm.refreshable import Refreshable
```
Comment (laimonassutkus) [inline b_lambda_layer_common/source/python/b_lambda_layer_common/cache/data_cache.py:12]: Importing from ssm directory does not make any sense. Write a TODO comment that "Refreshable" class should be moved somewhere else.

File: b_lambda_layer_common/source/python/b_lambda_layer_common/cache/data_cache.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
+import logging
+from typing import Any, Callable, Set, Union
+
+logger = logging.getLogger(__name__)
+
+try:
+    from b_lambda_layer_common.exceptions.container.internal_error import InternalError
+    from b_lambda_layer_common.ssm.refreshable import Refreshable
```
Comment (laimonassutkus) [inline b_lambda_layer_common/source/python/b_lambda_layer_common/cache/data_cache.py:8]: Importing from ssm directory does not make any sense. Write a TODO comment that "Refreshable" class should be moved somewhere else.

File: b_lambda_layer_common/source/python/b_lambda_layer_common/cache/data_cache.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        if self.should_refresh():
+            logger.info('Cached data should be reset.')
+            self.refresh()
+
+        if not isinstance(pointer, (str, set)):
+            raise InternalError('Type of pointer value must be str or set.')
+
+        return self.__get_data(pointer) or self.__add_data(pointer, func, *args, **kwargs)
+
+    def __get_data(self, pointer: Union[str, Set[str]]) -> Any:
```
Comment (laimonassutkus) [inline b_lambda_layer_common/source/python/b_lambda_layer_common/cache/data_cache.py:59]: Add docstrings.

File: b_lambda_layer_common/source/python/b_lambda_layer_common/cache/data_cache.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        elif isinstance(pointer, set) and pointer.issubset(self.__cache):
+            data = {key: self.__cache[key] for key in pointer}
+        else:
+            data = None
+
+        logger.info('Using cached data.' if data else 'Data not in cache.')
+
+        return data
+
+    def __add_data(self, pointer: Union[str, Set[str]], func: Callable, *args, **kwargs) -> Any:
```
Comment (laimonassutkus) [inline b_lambda_layer_common/source/python/b_lambda_layer_common/cache/data_cache.py:71]: Add docstrings.

**Review by laimonassutkus — CHANGES_REQUESTED** [review]

**Review by laimonassutkus — APPROVED** [review]

**Pull request merged**